qos/evm/endpoint.go

```diff
@@ -73,28 +125,25 @@ func (e *endpoint) ApplyObservation(obs *qosobservations.EVMEndpointObservation)
 	}
 
 	if blockNumberResponse := obs.GetBlockNumberResponse(); blockNumberResponse != nil {
-		// base 0: use the string's prefix to determine its base.
-		parsedBlockNumber, err := strconv.ParseUint(blockNumberResponse.GetBlockNumberResponse(), 0, 64)
-		// The endpoint returned an invalid response to an `eth_blockNumber` request.
-		// Explicitly set the parsedBlockNumberResponse to a zero value as the ParseUInt does not guarantee returning a 0 on all error cases.
-		if err != nil {
-			zero := uint64(0)
-			e.parsedBlockNumberResponse = &zero
+		e.parsedBlockNumberResponse = parseBlockNumberResponse(blockNumberResponse.GetBlockNumberResponse())
+		return true
+	}
+
+	if shouldPerformArchivalCheck {
```

Comment:
Whenever I start seeing `if flagForFeature {` in more than one place in the codebase, it creates a bit of a 👃 and gets my 🕷️ senses tingling.

I don't have a better suggestion (yet), but we could potentially have a `ApplyObservaionArchival` function and call that from within this one.

Wdyt?

Reply:
this has been fixed.